File: src/main/java/com/yinyang/project/entity/Report.java
package com.yinyang.project.entity;

import com.yinyang.project.dto.*;
import lombok.Getter;
import lombok.Setter;


@Setter
@Getter
public class Report {
    private TrendDataResponse donationAmountTrend;
    private TrendDataResponse donationCountTrend;
    private TopDataResponse<TopData> donationTopCategories;
    private TrendDataResponse userAccountGrowthTrend;
    private MultiTrendDataResponse userAccountWithProfileGrowthTrend;
    private Range range;

    public enum Range {
        DAILY, WEEKLY, MONTHLY
    }
}

File: frontend/src/api/report.js
import request from '@/utils/request.js'

export const getDailyReportService = (size) => {
    return request.get("/report/daily?size=" + size)
}

export const getWeeklyReportService = (size) => {
    return request.get("/report/weekly?size=" + size)
}

export const getMonthlyReportService = (size) => {
    return request.get("/report/monthly?size=" + size)
}